lib/iris/experimental/ugrid/utils.py

```diff
@@ -0,0 +1,322 @@
+
+"""
+from typing import AnyStr, Iterable
+
+import dask.array as da
+import numpy as np
+
+from iris.cube import Cube
+
+
+def recombine_regions(
+    full_mesh_cube: Cube,
+    region_cubes: Iterable[Cube],
+    index_coord_name: AnyStr = "i_mesh_index",
+) -> Cube:
+    """
+    Put data from regional sub-meshes back onto the original full mesh.
+
+    The result is a region_cube identical to 'full_mesh_cube', but with its data
+    replaced by a combination of data from the provided 'region_cubes'.
+    The result metadata, including name and units, are also replaced by those
+    of the 'region_cubes' (which must all be the same).
+
+    Args:
+
+    * full_mesh_cube
+        Describes the full mesh and mesh-location to which the region data
+        refers, and acts as a template for the result.
+        Must have a :class:`~iris.experimental.ugrid.mesh.Mesh`.
+
+    * region_cubes
+        Contain data on a subset of the 'full_mesh_cube' mesh locations.
+        The region cubes do not need to have a mesh.  There must be at least
+        1 of them, to determine the result phenomenon.
+        Their shapes and dimension-coords must all match those of
+        'full_mesh_cube', except in the mesh dimension, which can have
+        different sizes between the regions, and from the 'full_mesh_cube'.
+        The mesh dimension of each region cube must have a 1-D coord named by
+        'index_coord_name'.  Although these region index coords can vary in
+        length, they must all have matching metadata (names, units and
+        attributes), and must also match the coord of that name in the
+        'full_mesh_cube', if there is one.
+        The ".points" values of the region index coords specify, for each
+        datapoint, its location in the original mesh -- i.e. they are indices
+        into the relevant mesh-location dimension.
+
+    * index_coord_name
+        Coord name of the index coords in each region cubes, containing the
+        mesh location indices.
+
+    Result:
+
+    * result_cube
+        An unstructured region_cube identical to 'full_mesh_cube', and with the
+        same mesh and location, but with its data and ".metadata" replaced by
+        that from the 'region_cubes'.
+        Where regions overlap, the result data comes from the last-listed of the
+        original region cubes which contain that location.
+        Where no region contains a datapoint, it will be masked in the result.
+        HINT: alternatively, values covered by no region can be taken from the
+        original 'full_mesh_cube' data, if 'full_mesh_cube' is *also* passed
+        as the first of the 'region_cubes'.
+
+    """
+    if not region_cubes:
+        raise ValueError("'region_cubes' must be non-empty.")
+
+    mesh_dim = full_mesh_cube.mesh_dim()
+    if mesh_dim is None:
+        raise ValueError("'full_mesh_cube' has no \".mesh\".")
+
+    # Check the basic required properties of the input.
```

Comment:
NOTE: this is not wanted, since I changed my mind about the 'full_mesh' having to have a dim-coord on the mesh.
Now removed :